src/utils/settings.cpp

```diff
@@ -202,7 +202,7 @@ QString mp::Settings::get_client_settings_file_path() // idem
 void multipass::Settings::set_aux(const QString& key, QString val) // work with a copy of val
 {
     // TODO we should have handler callbacks instead
-    if (key == petenv_key && !mp::utils::valid_hostname(val.toStdString()))
+    if (key == petenv_key && !val.isEmpty() && !mp::utils::valid_hostname(val.toStdString()))
```

Comment:
OK, so only this change for now. Would you mind squashing all the commits so far accordingly? Thanks.